### PR TITLE
Don't catch InterruptedException in handleCompilationError

### PR DESCRIPTION
**Pull request by azolotko**
In combination with https://github.com/sbt/sbt/pull/5450, this PR addresses https://github.com/sbt/sbt/pull/5446#issuecomment-589985382

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=develop&zinc=pull/742/head&io=develop&librarymanagement=develop&util=develop&website=develop&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

File: zinc/src/main/scala/sbt/internal/inc/IncrementalCompilerImpl.scala [hunk 1]
```diff
@@ -167,7 +167,8 @@ class IncrementalCompilerImpl extends IncrementalCompiler {
     try {
       compilerRun
     } catch {
-      case e: CompileFailed => throw e // just ignore
+      case e: CompileFailed        => throw e // just ignore
+      case e: InterruptedException => throw e // just ignore
```
Comment (eed3si9n) [inline zinc/src/main/scala/sbt/internal/inc/IncrementalCompilerImpl.scala:171]: Good catch!

**Review by eed3si9n — APPROVED** [review]

File: zinc/src/main/scala/sbt/internal/inc/IncrementalCompilerImpl.scala (hunk 1 above)
Comment (eed3si9n) [reply zinc/src/main/scala/sbt/internal/inc/IncrementalCompilerImpl.scala:171]: Or I should say nice throw :)

**Pull request merged**